docs/query-builder.txt

```diff
@@ -9,588 +9,1116 @@ Query Builder
+         ...
+      ]
 
-The usage is the same as ``whereMonth`` / ``whereDay`` / ``whereYear`` / ``whereTime``
+.. _laravel-query-builder-null:
+
+Null or Missing Values Example
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following example shows how to use the ``whereNull()``
+query builder method to retrieve documents from the
+``movies`` collection that omit a ``runtime`` value
+or field.
+
+.. literalinclude:: /includes/query-builder/QueryBuilderTest.php
+   :language: php
+   :dedent:
+   :start-after: begin query whereNull
+   :end-before: end query whereNull
 
-**Advanced wheres**
+.. _laravel-query-builder-wherein:
 
-.. code-block:: php
+One or More Values of a Set Example
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following example shows how to use the ``whereIn()``
+query builder method to retrieve documents from the
+``movies`` collection that match at least one of the
+``title`` values in the specified set.
+
+.. io-code-block::
+   :copyable: true
 
-   $users =
-       User::where('name', 'John')
-           ->orWhere(function ($query) {
-               return $query
-                   ->where('votes', '>', 100)
-                   ->where('title', '<>', 'Admin');
-           })->get();
+   .. input:: /includes/query-builder/QueryBuilderTest.php
+      :language: php
+      :dedent:
+      :start-after: begin query whereIn
+      :end-before: end query whereIn
 
-**orderBy**
+   .. output::
+      :language: json
+      :visible: false
 
-.. code-block:: php
+      [
+         { "title": "Toy Story", "year": 1995, "runtime": 81, ... },
+         { "title": "Johnny English", "year": 2003, "runtime": 87, ... },
+         { "title": "Shrek 2", "year" 2004, "runtime": 93, ... },
+         ...
+      ]
 
-   $users = User::orderBy('age', 'desc')->get();
+.. _laravel-query-builder-wheredate:
 
-**Offset & Limit (skip & take)**
+Match Dates Example
+^^^^^^^^^^^^^^^^^^^
 
-.. code-block:: php
+The following example shows how to use the ``whereDate()``
+query builder method to retrieve documents from the
+``movies`` collection that match the specified date of
+``2010-1-15`` in the ``released`` field:
 
-   $users =
-       User::skip(10)
-           ->take(5)
-           ->get();
+.. literalinclude:: /includes/query-builder/QueryBuilderTest.php
+   :language: php
+   :dedent:
+   :start-after: begin query whereDate
+   :end-before: end query whereDate
+
+.. _laravel-query-builder-pattern:
 
-**groupBy**
+Text Pattern Match Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Selected columns that are not grouped will be aggregated with the ``$last``
-function.
+The following example shows how to use the ``like`` query operator
+with the ``where()`` query builder method to retrieve documents from the
+``movies`` collection by using a specified text pattern.
+
+Text patterns can contain text mixed with the following
+wildcard characters:
+
+- ``%`` which matches zero or more characters
+- ``_`` which matches a single character
```

Comment:
Q: if the `_` matches a single character, why is Spider Woman in the results?

Reply:
> Q: if the `_` matches a single character, why is Spider Woman in the results?

Good question -- whitespace is a character in the context of SQL queries. The [Laravel documentation](https://laravel.com/docs/11.x/queries#where-any-all-clauses) omits any explanation of the wildcard characters that they use, but I thought it would be good to reiterate the SQL LIKE syntax.
Let me know if you think this needs further explanation.